.github/workflows/horizon.yml

```
@@ -91,7 +91,19 @@ jobs:
       name: Setup Captive Core Remote Storage
       run: echo 'HORIZON_INTEGRATION_TESTS_CAPTIVE_CORE_USE_DB=true' >> $GITHUB_ENV
 
-    - run: go test -race -timeout 25m -v ./services/horizon/internal/integration/...
+    - name: Build Horizon reproducible build
+      run: |
+        go build -v -trimpath -buildvcs=false ./services/horizon
+
+    - name: Save Horizon binary and integration tests source hash to cache
+      id: horizon_binary_tests_hash
+      uses: actions/cache@v3
+      with:
+        path: ./horizon
+        key: horizon-hash-${{ hashFiles('./horizon') }}-${{ hashFiles('./services/horizon/internal/integration/**') }}-${{ env.PROTOCOL_19_CORE_DOCKER_IMG }}-${{ env.PROTOCOL_18_CORE_DOCKER_IMG }}
+
+    - if: ${{ steps.horizon_binary_tests_hash.outputs.cache-hit != 'true' }}
```

I suspect you can do the following, but not big deal.
```suggestion
    - if: !steps.horizon_binary_tests_hash.outputs.cache-hit
```

---

I found it in the docs and I'm not sure if you GH will auto convert true string to bool so I'll just leave it as it is.